processor/src/lib.rs

```diff
@@ -106,8 +114,17 @@ pub fn execute(
 
 /// Returns an iterator which allows callers to step through the execution and inspect VM state at
 /// each execution step.
-pub fn execute_iter(program: &Program, inputs: &ProgramInputs) -> VmStateIterator {
-    let mut process = Process::new_debug(program.kernel(), inputs.clone());
+pub fn execute_iter<ADV, STK, V>(
+    program: &Program,
+    advice: ADV,
+    stack_init: STK,
+) -> VmStateIterator<ADV>
```

Comment:
Similar comment to the one above: I'd prefer to keep external interfaces unchanged in this PR and change them in subsequent PRs.

Reply:
Done!